server/schedule/placement/fit_test.go

```diff
@@ -38,131 +38,120 @@ func (s *testFitSuite) makeStores() map[uint64]*core.StoreInfo {
 						"zone": fmt.Sprintf("zone%d", zone),
 						"rack": fmt.Sprintf("rack%d", rack),
 						"host": fmt.Sprintf("host%d", host),
+						"id":   fmt.Sprintf("id%d", x),
 					}
-					stores[id] = core.NewStoreInfoWithLabel(id, 0, labels)
+					stores.SetStore(core.NewStoreInfoWithLabel(id, 0, labels))
 				}
 			}
 		}
 	}
 	return stores
 }
 
-func (s *testFitSuite) TestFitByLocation(c *C) {
-	stores := s.makeStores()
+// example: "1111_leader,1234,2111_learner"
+func (s *testFitSuite) makeRegion(def string) *core.RegionInfo {
+	var regionMeta metapb.Region
+	var leader *metapb.Peer
+	for _, peerDef := range strings.Split(def, ",") {
+		role, idStr := Follower, peerDef
+		if strings.Contains(peerDef, "_") {
+			splits := strings.Split(peerDef, "_")
+			idStr, role = splits[0], PeerRoleType(splits[1])
+		}
+		id, _ := strconv.Atoi(idStr)
+		peer := &metapb.Peer{Id: uint64(id), StoreId: uint64(id), IsLearner: role == Learner}
+		regionMeta.Peers = append(regionMeta.Peers, peer)
+		if role == Leader {
+			leader = peer
+		}
+	}
+	return core.NewRegionInfo(&regionMeta, leader)
+}
 
-	type Case struct {
-		// peers info
-		peerStoreID []uint64
-		peerRole    []PeerRoleType // default: all Followers
-		// rule
-		locationLabels string       // default: ""
-		count          int          // default: len(peerStoreID)
-		role           PeerRoleType // default: Voter
-		// expect result:
-		expectedPeers []uint64 // default: same as peerStoreID
+// example: "3/voter/zone=zone1,rack=rack2/zone,rack,host"
+//       count role constraints location_labels
+func (s *testFitSuite) makeRule(def string) *Rule {
+	var rule Rule
+	splits := strings.Split(def, "/")
+	rule.Count, _ = strconv.Atoi(splits[0])
+	rule.Role = PeerRoleType(splits[1])
+	// only support k=v type constraint
+	for _, c := range strings.Split(splits[2], ",") {
+		if c == "" {
+			break
+		}
+		kv := strings.Split(c, "=")
+		rule.LabelConstraints = append(rule.LabelConstraints, LabelConstraint{
+			Key:    kv[0],
+			Op:     "in",
+			Values: []string{kv[1]},
```

Comment:
It's ok for me to only support `in` operator here, but could you support multi value for the `in` operator instead of only one value? I have some unit tests need this feature.

Reply:
good idea.

Reply:
@Yisaer addressed